test/e2e/framework/helper/certificates.go

```diff
@@ -71,15 +71,9 @@ func (h *Helper) waitForCertificateCondition(client clientset.CertificateInterfa
 	if pollErr != nil && certificate != nil {
 		log.Logf("Failed waiting for certificate %v: %v\n", name, pollErr.Error())
 
-		if len(certificate.Status.Conditions) > 0 {
-			log.Logf("Observed certificate conditions:\n")
-			for _, cond := range certificate.Status.Conditions {
-				log.Logf("- Last Status: '%s' Reason: '%s', Message: '%s'\n", cond.Status, cond.Reason, cond.Message)
-			}
-		}
+		log.Logf("Certificate:\n")
+		h.describeCMObject(certificate)
 
-		log.Logf("Certificate description:\n")
-		h.Kubectl(certificate.Namespace).DescribeResource("certificate", name)
 		log.Logf("Order and challenge descriptions:\n")
 		h.Kubectl(certificate.Namespace).Describe("order", "challenge")
```

Comment:
Perhaps we can re-use some of the code from https://github.com/jetstack/cert-manager/blob/master/cmd/ctl/pkg/status/certificate/certificate.go here and get even more diagnostic info. If not in this PR, we could do it in a follow up PR if you agree that it'd be useful.

Reply:
I agree, there are possible improvements to be made here.